### ISSUE TITLE
Consider relaxing ETC1S I-Frame predictor restriction

### ISSUE THREAD

**Issue by lexaknyazev** [BinomialLLC/basis_universal#222]
Currently, still ETC1S slices can use any of 4 available predictors (left, upper, upper-left, and none).

Let's say there's an array of ETC1S slices that were compressed as independent still images sharing the same codebooks and stored in the same file. Semantically, they could be easily turned into an I-Frame-only video by adjusting the header without recompression, but the restriction on the upper-left predictor usage prevents that.

Moreover, this restriction slightly reduces the efficiency of I-Frame encoding compared to still images although there's no technical difference between them.

Would it be possible to relax this restriction by saying that predictor usage for I-Frames is the same as for still images, while for P-Frames the value of 2 is redefined?

This change is backwards-compatible, i.e. already compressed basis files won't be affected.

**Comment by richgel999** [comment]
> Would it be possible to relax this restriction by saying that predictor usage for I-Frames is the same as for still images, while for P-Frames the value of 2 is redefined?

Yes - this is something that's been in the back of my mind. It currently works the way it does on I-frames only for simplicity. It makes perfect sense to fix this and it'll boost compression. It's on the list - it'll involve some moderately tricky changes to the backend and simple changes to the transcoder. The ETC1S spec will need some simple changes/additions, but I don't think it'll be a big deal.
